sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -373,6 +373,11 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val STARJOIN_OPTIMIZATION = SQLConfigBuilder("spark.sql.starJoinOptimization")
+    .doc("When true, it enables join reordering based on star schema detection. ")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
Do you suggest to have another internal config that will be enabled for testing purposes?